packages/next/src/executors/build/build.impl.ts

```diff
@@ -31,7 +31,11 @@ export default async function buildExecutor(
   const hasReact18 =
     reactDomVersion &&
     gte(checkAndCleanWithSemver('react-dom', reactDomVersion), '18.0.0');
-  (process.env as any).__NEXT_REACT_ROOT ||= hasReact18 ? 'true' : undefined;
+  if (hasReact18) {
+    (process.env as any).__NEXT_REACT_ROOT ||= 'true';
+  } else {
+    delete (process.env as any).__NEXT_REACT_ROOT;
+  }
```

Comment:
Thanks for the quick change @puku0x :)

Just wanted to add that the if statement by itself should be sufficient:

```
 if (hasReact18) {
    (process.env as any).__NEXT_REACT_ROOT ||= 'true';
}
```

Unless you guys want to make sure `__NEXT_REACT_ROOT` isn't settable in userland, which might or might not make sense.

Reply:
Sounds great. https://github.com/nrwl/nx/pull/11749/commits/708d6fbfe431d585d5e437d7274edc9bb45f8526